### PR TITLE
HAAR-161: filter by prison type code

### PR DESCRIPTION
**Pull request by neilmendum**
https://dsdmoj.atlassian.net/browse/HAAR-162

### CODE REVIEW

File: src/main/resources/db/migration/V1_12__fix_prison_type.sql [hunk 1]
```diff
@@ -0,0 +1 @@
+UPDATE PRISON_TYPE SET TYPE = 'YOI' WHERE PRISON_ID = 'ISI';
```
Comment (neilmendum) [inline src/main/resources/db/migration/V1_12__fix_prison_type.sql:1]: We accidentally set this to `/YO` in `V1_10__insert_prison_types.sql`

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonregister/model/PrisonFilter.kt [hunk 2]
```diff
@@ -21,13 +22,19 @@ class PrisonFilter(
...
+      andBuilder.add(prisonTypesPredicate)
+    }
     if (!textSearch.isNullOrBlank()) {
       val orBuilder = ImmutableList.builder<Predicate>()
       orBuilder.add(textSearchWildcardAndIgnoreCasePredicate(root, cb, "prisonId"))
       orBuilder.add(textSearchWildcardAndIgnoreCasePredicate(root, cb, "name"))
       andBuilder.add(cb.or(*orBuilder.build().toTypedArray()))
     }
     query.orderBy(cb.asc(root.get<Any>("prisonId")))
+    query.distinct(true)
```
Comment (neilmendum) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisonregister/model/PrisonFilter.kt:37]: The join with `prison_type` meant the query was pulling back duplicate records so included `distinct`

Comment (karenmillermoj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/prisonregister/model/PrisonFilter.kt:37]: It is probably worth adding a test for this scenario (to ensure no duplications)

File: src/test/kotlin/uk/gov/justice/digital/hmpps/prisonregister/model/PrisonRepositoryTest.kt [hunk 3]
```diff
@@ -117,9 +117,34 @@ class PrisonRepositoryTest {
...
 
     @Test
-    fun `should find prisons by active , text search , male flag`() {
-      val prisonsByMultipleFields = prisonRepository.findAll(PrisonFilter(active = true, textSearch = "vei", genders = listOf(Gender.MALE)))
-      assertThat(prisonsByMultipleFields.first()).isEqualTo(Prison("VEI", "The Verne (HMP)", active = true, male = true))
+    fun `should find prisons by prison type`() {
+      val hmpPrisons = prisonRepository.findAll(PrisonFilter(prisonTypeCodes = listOf(Type.HMP)))
+      assertThat(hmpPrisons).hasSizeGreaterThan(100)
+
+      val yoiAndIrcPrisons = prisonRepository.findAll(PrisonFilter(prisonTypeCodes = listOf(Type.YOI, Type.IRC)))
```
Comment (karenmillermoj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/prisonregister/model/PrisonRepositoryTest.kt:124]: nice!

**Review by karenmillermoj — APPROVED** [review]
lgtm.  Just the extra repository test would be good

**Pull request merged**